serve.go

```diff
@@ -24,19 +29,86 @@ func (s *Server) Run() error {
 // It also generates the OpenAPI spec and outputs it to a file, the UI, and a handler (if enabled).
 func (s *Server) RunTLS(certFile, keyFile string) error {
 	s.isTLS = true
-
-	s.setup()
-	return s.Server.ListenAndServeTLS(certFile, keyFile)
+	if err := s.setup(certFile, keyFile); err != nil {
+		return err
+	}
```

Comment:
Can't you call WithTLSListener here?

It would allow to pass certFile and keyFile to setup, and the check on isTLS

Reply:
The code was significantly simplified based on the observation. Can you help check if there’s any room for improvement? Thank you very much!

Reply:
Great.

I'll open a PR to your fork

Reply:
I opened https://github.com/Ignaciojeria/fuego/pull/1